libraries/AP_Compass/AP_Compass.h

```diff
@@ -480,7 +487,31 @@ friend class AP_Compass_Backend;
         // accumulated samples, protected by _sem, used by AP_Compass_Backend
         Vector3f accum;
         uint32_t accum_count;
-    } _state[COMPASS_MAX_INSTANCES];
+        // We only copy persistent params
+        void copy_from(const mag_state& state) {
+            external.set_and_save(state.external);
+            orientation.set_and_save(state.orientation);
+            offset.set_and_save(state.offset);
+            diagonals.set_and_save(state.diagonals);
+            offdiagonals.set_and_save(state.offdiagonals);
+            scale_factor.set_and_save(state.scale_factor);
+            dev_id.set_and_save(state.dev_id);
+            motor_compensation.set_and_save(state.motor_compensation);
+            expected_dev_id = state.expected_dev_id;
+        }
+    };
+
+    RestrictIDTypeArray<mag_state, COMPASS_MAX_INSTANCES+1, StateIndex> _state;
+
+    StateIndex _get_state_id(Priority priority) const;
```

Comment:
can we get some comments added around these new members and methods to clarify their purpose?

Reply:
will do